src/ar_bundles.erl

```diff
@@ -418,14 +418,24 @@ deserialize_item(Binary) ->
             tags = Tags,
             data = Data,
             data_size = byte_size(Data)
-        })
+        }),
+        Opts
     ).
 
-maybe_unbundle(Item) ->
+maybe_unbundle(Item, Opts) ->
     case dev_arweave_common:type(Item) of
-        list -> unbundle_list(Item);
-        binary -> Item;
-        map -> unbundle_map(Item)
+        list ->
+            UnbundleBundles = hb_opts:get(<<"unbundle_bundles">>, true, Opts),
+            case UnbundleBundles of
+                true -> 
+                    unbundle_list(Item);
+                false ->
+                    Item
+            end;
```

Comment:
This is the core logic, everything else is making sure `Opts` is given to the functions.